### DISCUSSION TITLE
Row level security, adding a JWT property?

### DISCUSSION THREAD

**Discussion by sitefinitysteve** [supabase/supabase#22675] [Questions] [answered]
TLDR: I need my app to just shoot up the users saved JSON settings.  

I have their unique device UUID and I'm storing that in the settings table, can query it just fine. But the question is how do I add row level security such that only they can see\update the record with their UUID?  The AI generation seems to want to add a property to the jwt, but how on my js client do I add to that? There doesn't seem to be any methods off the supabase client object to do that (or am I missing them)

To be clear there's NO personal identifiable information in the row, it's just basic settings for the app...

**Comment by GaryAustin1** [comment, marked as answer]
I would pass it in as a header when you createClient and you can access all the headers in RLS.

https://postgrest.org/en/v12/references/transactions.html#request-headers-cookies-and-jwt-claims

**Reply by sitefinitysteve** [reply]
Perfect I'll try thank you!

...got AI'd again i suppose

**Comment by sitefinitysteve** [comment]
Going to just drop this here incase anyone else needs it going forward

( SELECT (((current_setting('request.headers'::text, true))::json ->> 'device_id'::text) = ("userdata".device_id)::text))
